src/lib/frontend/Main.svelte

```diff
@@ -7,13 +7,20 @@
   let shoppingList: Product[] = [];
 
   function addToMainList(event) {
-    console.log("Received", event)
-    shoppingList.push(event.detail.value);
-    shoppingList = shoppingList
+    console.log("Received", event);
+
+    let product: Product = event.detail.value;
+    if (shoppingList.includes(product)) {
+      alert("Produkt " + product.name + " jest już w Twojej liście zakupów!");
+    } else {
+      shoppingList.push(product);
+      shoppingList = shoppingList;
+    }
   }
+  
   function removeFromShoppingList(i: number) {
     shoppingList.splice(i, 1);
-    shoppingList = shoppingList
+    shoppingList = shoppingList;
```

Comment:
Co to za dziwna linie?

Reply:
Zformatowanie pliku w VS Code mi ją dodało :eyes: